src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,53 @@
 package core.basesyntax.service;
 
+import core.basesyntax.RegistrationException;
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMAL_LENGTH = 6;
+    private static final int MINIMAL_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        return storageDao.add(checkUserValidation(user));
+    }
+
+    public User checkUserValidation(User user) throws RegistrationException {
+        if (user == null) {
+            throw new RegistrationException("Invalid data");
+        }
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login can't be null");
+        }
+        if (user.getLogin().isEmpty()) {
+            throw new RegistrationException("Login can't be empty");
+        }
+        if (user.getPassword() == null) {
+            throw new RegistrationException("Password can't be null");
+        }
```

Comment:
The check for `user.getPassword() == null` and `user.getPassword().isEmpty()` can be combined into one check using `isBlank()` if you are using Java 11 or later.